### PR TITLE
show spinner until chat image arrives

### PR DESCRIPTION
**Pull request by zanderz**
This is the case where the sender uploads fine but the receiver can't reach our servers to get it.
Also requested by @mmaxim  in https://github.com/keybase/client/pull/11516

I don't know how to block specific urls to test on mobile.

image

### CODE REVIEW

File: shared/chat/conversation/messages/attachment/image/index.js [hunk 1]
```diff
@@ -55,12 +56,16 @@ class ImageAttachment extends React.PureComponent<Props, State> {
...
             {
               height: this.props.height,
               width: this.props.width,
             },
           ])}
         >
+          {!this.state.loaded && (
+             <ProgressIndicator style={{margin: 'auto'}} />
+          )}
           {!!this.props.path && (
```
Comment (songgao) [inline shared/chat/conversation/messages/attachment/image/index.js]: If looks like if we do `!this.state.loaded ? ....`, this block can just be part of what's after `:`. Then we don't need `display: none` and can get rid of `opacity` below.

File: shared/chat/conversation/messages/attachment/image/index.js [hunk 2]
```diff
@@ -55,12 +56,16 @@ class ImageAttachment extends React.PureComponent<Props, State> {
         <Box
           style={collapseStyles([
             styles.loading,
+            !this.state.loaded ? styles.spinner : {},
```
Comment (songgao) [inline shared/chat/conversation/messages/attachment/image/index.js]: Hmm I wonder if styles.spinner can just go into `styles.loading`?

Comment (chrisnojima) [inline shared/chat/conversation/messages/attachment/image/index.js]: you can just do 
```
!this.state.loaded && styles.spinner
```

File: shared/chat/conversation/messages/attachment/image/index.js [hunk 3]
```diff
@@ -55,12 +56,16 @@ class ImageAttachment extends React.PureComponent<Props, State> {
...
             {
               height: this.props.height,
               width: this.props.width,
             },
           ])}
         >
+          {!this.state.loaded && (
+             <ProgressIndicator style={{margin: 'auto'}} />
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/attachment/image/index.js]: i doubt this works in mobile

File: shared/chat/conversation/messages/attachment/image/index.js [hunk 4]
```diff
@@ -72,6 +77,7 @@ class ImageAttachment extends React.PureComponent<Props, State> {
                   opacity: this.state.loaded ? 1 : 0,
                   width: this.props.width,
                 },
+                this.state.loaded ? null : {display: 'none'},
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/attachment/image/index.js]: also this

**Comment by zanderz** [comment]
thanks guys, I jumped the gun, it doesn't work on mobile, sorry.

**Comment by zanderz** [comment]
spinner still not vertically centered on mobile, but better than nothing.

**Pull request merged**